lib/model.js

```diff
@@ -2172,8 +2172,8 @@ class Model {
       hooks: true
     }, Utils.cloneDeep(options || {}));
 
-    const createdAtAttr = this._timestampAttributes.createdAt;
-    const updatedAtAttr = this._timestampAttributes.updatedAt;
+    const createdAtAttr = this.rawAttributes[this._timestampAttributes.createdAt].field || this._timestampAttributes.createdAt;
+    const updatedAtAttr = this.rawAttributes[this._timestampAttributes.updatedAt].field || this._timestampAttributes.updatedAt;
```

Comment:
I suspect this could fail when timestamps are disabled, please add a test for that case